testplan/common/remote/remote_resource.py

```diff
@@ -400,31 +423,64 @@ def _prepare_workspace(self, exist_on_remote: bool) -> None:
             )
             self._transfer_data(
                 # join with "" to add trailing "/" to source
-                # this will copy everything under local import path to to testplan_lib
+                # this will copy everything under local workspace path to fetched_workspace
                 source=os.path.join(self._workspace_paths.local, ""),
                 target=self._workspace_paths.remote,
                 remote_target=True,
                 exclude=self.cfg.workspace_exclude,
             )
+
+            if IS_WIN:
+                return
+
             self.logger.info(
                 "%s: creating symlink to imitate local workspace path on %s, "
                 "pointing to runpath/fetched_workspace",
                 self,
                 self.ssh_cfg["host"],
             )
-            self._execute_cmd_remote(
+            r, w = os.pipe()
+            rmt_non_existing = None
+            if 0 == self._execute_cmd_remote(
+                cmd="/bin/bash -c "
+                + shlex.quote(
+                    'e=""; for i in '
+                    + " ".join(
+                        map(
+                            shlex.quote,
+                            self._workspace_paths.local.split(os.sep)[1:],
+                        )
+                    )
+                    + '; do e+="/${i}"; if [ ! -e "$e" ]; then echo "$e"; break; fi; done'
+                ),
+                label="imitate local workspace path on remote - detect non-existing",
+                stdout=os.fdopen(w),
+                check=False,  # XXX: bash might not be there
+            ):
+                rmt_non_existing = os.fdopen(r).read().strip() or None
+            if 0 == self._execute_cmd_remote(
                 cmd=mkdir_cmd(os.path.dirname(self._workspace_paths.local)),
                 label="imitate local workspace path on remote - mkdir",
                 check=False,  # just best effort
-            )
-            self._execute_cmd_remote(
+            ) and 0 == self._execute_cmd_remote(
                 cmd=link_cmd(
                     path=self._workspace_paths.remote,
                     link=self._workspace_paths.local,
                 ),
                 label="imitate local workspace path on remote - ln",
                 check=False,  # just best effort
-            )
+            ):
+                # NOTE: we shall always remove created symlink
+                self._dangling_remote_fs_obj = (
```

Comment:
we can though, just remove the symlink we created, at least to avoid circular symlink problem for user. Take care of more complicated use case later.

Reply:
sure thing, will convert some lines into comments